BCnEnc.Net/Encoder/BcEncoder.cs

```diff
@@ -50,247 +72,136 @@ public BcEncoder(CompressionFormat format = CompressionFormat.Bc1)
 		/// <summary>
 		/// Encodes all mipmap levels into a ktx or a dds file and writes it to the output stream asynchronously.
 		/// </summary>
-		/// <param name="inputPixelsRgba">The input image rgba data.</param>
-		/// <param name="width">The width of the input image.</param>
-		/// <param name="height">The height of the input image.</param>
+		/// <param name="input">The input to encode represented by a <see cref="Memory{T}"/>.</param>
+		/// <param name="width">The width of the image.</param>
+		/// <param name="height">The height of the image.</param>
+		/// <param name="format">The pixel format the given data is in.</param>
 		/// <param name="outputStream">The stream to write the encoded image to.</param>
 		/// <param name="token">The cancellation token for this operation. Can be default, if the operation is not asynchronous.</param>
-		public Task EncodeToStreamAsync<T>(Memory<T> inputPixelsRgba, int width, int height, Stream outputStream, CancellationToken token = default) where T : unmanaged
+		public Task EncodeToStreamAsync(Memory<byte> input, int width, int height, PixelFormat format, Stream outputStream, CancellationToken token = default)
```

Comment:
I haven't looked too deeply at the usage of `input`, but I noticed all throughout the code inputs are Memory<T>. If you are not modifying them it would be better to use ReadOnlyMemory<T> instead to show clearer intent.

Reply:
Ah, yes. That would indeed be better probably.

Reply:
Also, if possible, any synchronous APIs should prefer using `Span<T>` over `Memory<T>` (or the read only equivalent).

Refer to Rule 1 here:
https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines#usage-guidelines

Reply:
Thanks for this link. I haven't read these before

Reply:
I'm aware of those information and I decided against them for simple reasons:
1. The mip generation would ultimately need to revert to Memory<T> since you cannot retrun a Span<T> or Span2D<T> from a method
2. Going by point 1, to get a span into a memory, you would need to retrieve its underlying buffer array, which is only possible by using ToArray() on the span, which copies the content of the span to a new array. This is not very memory efficient.

By going with a Memory<T> we have less copying of buffers overall.

Reply:
Since there seem to be no further comments to this, I will resolve this conversation.